usr/src/uts/common/fs/zfs/zfs_vfsops.c

```diff
@@ -1023,14 +1023,6 @@ zfsvfs_setup(zfsvfs_t *zfsvfs, boolean_t mounting)
 {
 	int error;
 
-	/*
-	 * Check for a bad on-disk format version now since we
-	 * lied about owning the dataset readonly before.
-	 */
-	if (!(zfsvfs->z_vfs->vfs_flag & VFS_RDONLY) &&
-	    dmu_objset_incompatible_encryption_version(zfsvfs->z_os))
-		return (SET_ERROR(EROFS));
```

Comment:
Removing this check will cause the errata test to blow up because it will be able to mount datasets on the pool without the readonly option, which shouldn't be allowed.

Reply:
Actually, I went back to ZOL and aimed to re-port it, only to find it is not there. In fact, I can't find anything calling `dmu_objset_incompatible_encryption_version()` in ZOL. But, the goal is to be like ZOL's code in here, so I removed the test, like they have.

Reply:
I checked with caputi:
"everyone uses `dsl_dir_incompatible_encryption_version()`
the other one is just exported for external modules"
and our calls match that of ZOL, so it should work as intended.